outputter/writer.go

```diff
@@ -36,23 +35,26 @@ func (f *file) Handle(data string) error {
 	if err := f.validate(); err != nil {
 		return err
 	}
-	file, err := os.Create(f.Filename)
-	if err != nil {
-		return err
+
+	var w io.Writer
+	if len(f.Filename) == 0 {
+		w = os.Stdout
+	} else {
+		file, err := os.Create(f.Filename)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		w = file
 	}
-	defer file.Close()
 
-	w := bufio.NewWriter(file)
-	defer w.Flush()
+	writer := bufio.NewWriter(w)
+	defer writer.Flush()
```

Comment:
Something feels wrong here:  `f.validate()` checks that `f.ioWriter` is non-nil, but then you use a different writer altogether. This all feels very complicated for something that is essentially just writing a string (which may have been formatted as console output or JSON or whatever) to a file. I suspect there is at least one unnecessary level of abstraction here; I'm not sure why all these three outputters don't end up simply doing `io.WriteString()`. 

I also wonder if the `file` structure is helpful or just adding extra complexity. Could `Handle` be a direct part of the `Outputter` interface? Possibly better named as `Write`?